### DISCUSSION TITLE
Question about uv_tcp_t timeout.

### DISCUSSION THREAD

**Discussion by krishna116** [libuv/libuv#3633] [Q&A] [answered]
When a client use uv_tcp_t to connect a server,
how to set connect timeout? 
how to set every read timeout?
how to set every write timeout?
I find some people use a uv_timer_t to control the timeout, but it seems complicated, should I use some api like setsockopt? what is the best practice to handle read/write timeout?

Thank you very much.

**Comment by bnoordhuis** [comment, marked as answer]
Use a `uv_timer_t`. Mechanisms like `setsockopt(SO_RCVTIMEO)` don't work for non-blocking sockets.

**Reply by krishna116** [reply]
when I search libuv document about timeout, there is just a uv_timer_t, so it seems no other api or method  to handle timeout, thank you.
